lib/t/feeds.ex

```diff
@@ -97,42 +97,110 @@ defmodule T.Feeds do
         ) ::
           {[feed_profile], feed_cursor}
   def fetch_feed(user_id, location, gender, gender_preferences, count, feed_cursor) do
-    feed_profiles =
-      feed_profiles_q(user_id, gender, gender_preferences, feed_cursor)
-      |> limit(^count)
-      |> select([p], {p, distance_km(^location, p.location)})
-      |> Repo.all()
+    case feed_cursor do
+      nil -> save_eligible_feed_profiles_to_memory(user_id, location, gender, gender_preferences)
+      _ -> :ok
+    end
+
+    feed_profiles_with_cursor = feed_profiles_from_memory(user_id, feed_cursor, count)
 
     feed_cursor =
-      if last = List.last(feed_profiles) do
-        {feed_profile, _distance} = last
-        feed_profile.last_active
+      if last = List.last(feed_profiles_with_cursor) do
+        {cursor, _feed_profile} = last
+        cursor
       else
         feed_cursor
       end
 
+    feed_profiles =
+      feed_profiles_with_cursor
+      |> Enum.map(fn {_cursor, profile} ->
+        profile
+      end)
+
     {feed_profiles, feed_cursor}
   end
 
+  def save_eligible_feed_profiles_to_memory(user_id, location, gender, gender_preferences) do
+    table = String.to_atom("feed_#{user_id}")
+
+    case :ets.whereis(table) do
+      :undefined -> :ok
+      _ -> :ets.delete(table)
+    end
+
+    :ets.new(table, [:ordered_set, :named_table, read_concurrency: true])
```

Comment:
`read_concurrency: true` is only needed when the table is being read from different processes. In current implementation a single feed channel process is reading from the table. Also the behaviour is undefined when two processes read from the same table.